packages/api/lib/GranuleFilesCache.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * The authoritative source for granule files is the `files` property of a
+ * granule stored in the Granules table in DynamoDB. There are times when, given
+ * an S3 bucket and key, we need to determine what granule that object is
+ * associated with. Relying on the Granules table requires a full table scan,
+ * which is inefficient and relatively expensive. To make that easier, there is
+ * a Files table in DynamoDB. This table uses `bucket` and `key` as keys, and
+ * contains a `granuleId` property. This table is _not_ an authoritative source,
+ * though, and it's possible that it will not be in sync with the granules
+ * table.
+ */
+
+'use strict';
+
+const chunk = require('lodash.chunk');
+const flatten = require('lodash.flatten');
+const pick = require('lodash.pick');
+const pMap = require('p-map');
+const { isNil, noop } = require('@cumulus/common/util');
+const { dynamodb, dynamodbDocClient } = require('@cumulus/aws-client/services');
+const { isNonEmptyString } = require('@cumulus/common/string');
+
+/**
+ * Return the name of the Granule Files cache table
+ * @returns {string} the name of the DynamoDB table
+ */
+const cacheTableName = () => {
+  if (isNonEmptyString(process.env.FilesTable)) return process.env.FilesTable;
+  throw new Error('process.env.FilesTable is not set');
+};
+
+/**
+ * Perform a bulk-update of the Granule Files cache table.
+ *
+ * Objects in the `puts` array will be added to the cache or update an existing
+ * cache entry. `puts` objects must contain a `bucket`, `key`, and `granuleId`.
+ *
+ * Objects in the `deletes` array will be removed from the cache, and must
+ * contain a `bucket` and `key`.
+ *
+ * @param {Object} params
+ * @param {Array<Object>} params.puts - a list of new files to be added to the
+ *   cache
+ * @param {Array<Object>} params.deletes - a list of files to be removed from
+ *   the cache
+ */
+const batchUpdate = async (params = {}) => {
+  const { puts = [], deletes = [] } = params;
+
+  // Perform input validation
+  puts.forEach((put) => {
```

Comment:
You don't need to change it feels like this one and the next one could be extracted into a reusable function or also a utility function outside this file